CMakeLists.txt

```diff
@@ -533,10 +501,47 @@ set(CPACK_NSIS_INSTALLER_MUI_ICON_CODE "!define MUI_WELCOMEFINISHPAGE_BITMAP \\\
 
 include(CPack)
 
-if(NOT CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+# configure extras by default when building SFML directly, otherwise hide them
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+    set(SFML_CONFIGURE_EXTRAS_DEFAULT TRUE)
+else()
+    set(SFML_CONFIGURE_EXTRAS_DEFAULT FALSE)
+endif()
+sfml_set_option(SFML_CONFIGURE_EXTRAS ${SFML_CONFIGURE_EXTRAS_DEFAULT} BOOL "TRUE to configure extras, FALSE to ignore them")
```

Comment:
Just default this to true - The expectation is that the configuration is the same whether it's a subdirectory or not, then users can decide which targets they want to enable

Reply:
The main idea behind this PR is to get `add_subdirectory` (and thus `FetchContent`) to behave as similarly as `find_package`. I can see merits and supporting these use cases more.  
As such `SFML_CONFIGURE_EXTRAS` is a good compromise to hide additional options in those scenarios, while still providing the possibility to access the extra options.

Reply:
I won't be defaulting it true because the point of this PR is to not expose such things to downstream projects by default. It's expected for users to not want such things and exceptional if they do.